soh/src/code/z_parameter.c

```diff
@@ -5006,33 +4979,45 @@ void Interface_Draw(PlayState* play) {
 
         if (fullUi) {
             // Rupee Icon
-            if (CVar_GetS32("gHudColors", 1) == 2) {
-                //Custom Color is on but check if Dynamic Wallet is on.
-                if (CVar_GetS32("gDynamicWalletIcon", 0) != 0) {
-                    //if on let's use Dynamic Colors
-                    rColor.r = rupeeWalletColors[CUR_UPG_VALUE(UPG_WALLET)][0];
-                    rColor.g = rupeeWalletColors[CUR_UPG_VALUE(UPG_WALLET)][1];
-                    rColor.b = rupeeWalletColors[CUR_UPG_VALUE(UPG_WALLET)][2];
-                } else {
-                    //else use our custom color
-                    rColor = CVar_GetRGB("gCCRupeePrim", rColor_ori);
+            if (CVar_GetS32("gDynamicWalletIcon", 0)) {
+                switch (CUR_UPG_VALUE(UPG_WALLET)) {
+                    case 0:
```

Comment:
maybe out of scope but would be cool to have the different wallet upgrade values as an enum instead of magic numbers

Reply:
i'll probably do another magic number sweep at some point in the future (most likely just search for all switch/cases and try to replace the case magic numbers with enum values) this doesn't need to happen now